### ISSUE TITLE
Crash with multi-root workspace (pre-release)

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#3907]
Starting a multi-root workspace with the latest pre-release version crashes the language server. It worked fine for `2023.2.10`.
```
The Pylance server crashed 5 times in the last 3 minutes.
The server will not be restarted. See the output for more information.
```

### Crash log
```
[Info  - 7:07:24 PM] (19522) Pylance language server 2023.2.12 (pyright cca7588f) starting
[Info  - 7:07:24 PM] (19522) Server root directory: /Users/marc/.vscode/extensions/ms-python.vscode-pylance-2023.2.12/dist
[Info  - 7:07:24 PM] (19522) Starting service instance "<workspace_root>"
[Info  - 7:07:24 PM] (19522) Starting service instance "<other_folder>"
[Info  - 7:07:24 PM] (19522) Starting service instance "<remote_folder>"
[Info  - 7:07:25 PM] (19522) Starting service instance "<default>"
[Info  - 7:07:25 PM] (19522) No pyproject.toml file found.
[Info  - 7:07:25 PM] (19522) Setting pythonPath for service "<default>": "/.../<workspace_root>/venv/bin/python"
[Warn  - 7:07:25 PM] (19522) stubPath typings is not a valid directory.
[Info  - 7:07:25 PM] (19522) Assuming Python version 3.10
[Info  - 7:07:25 PM] (19522) Assuming Python platform Darwin
[Info  - 7:07:26 PM] (19522) Search paths for <default>
[Info  - 7:07:26 PM] (19522)   /Users/marc/.vscode/extensions/ms-python.vscode-pylance-2023.2.12/dist/typeshed-fallback/stdlib
[Info  - 7:07:26 PM] (19522)   typings
[Info  - 7:07:26 PM] (19522)   /Users/marc/.vscode/extensions/ms-python.vscode-pylance-2023.2.12/dist/typeshed-fallback/stubs/...
[Info  - 7:07:26 PM] (19522)   /Users/marc/.vscode/extensions/ms-python.vscode-pylance-2023.2.12/dist/bundled/stubs
[Info  - 7:07:26 PM] (19522)   /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10
[Info  - 7:07:26 PM] (19522)   /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/lib-dynload
[Info  - 7:07:26 PM] (19522)   /.../<workspace_root>/venv/lib/python3.10/site-packages
[Info  - 7:07:26 PM] (19522)   /.../<workspace_root>/<project>
[Info  - 7:07:26 PM] (19522) Searching for source files
[Info  - 7:07:26 PM] (19522) No source files found.
Error: spawn /.../<workspace_root>/venv/bin/python ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
...
<Same error repeated multiple times>
...
[Error - 7:07:41 PM] The Pylance server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
```

> **Note**
> `<workspace_root>`, `<other_folder>`, `<remote_folder>`, and `/.../` are placeholders inserted by me.

### Reference (last working version)
```
[Info  - 7:14:22 PM] (19667) Pylance language server 2023.2.10 (pyright 01758105) starting
[Info  - 7:14:22 PM] (19667) Server root directory: /Users/marc/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist
[Info  - 7:14:22 PM] (19667) Starting service instance "<workspace_root>"
[Info  - 7:14:22 PM] (19667) Starting service instance "<other_folder>"
[Info  - 7:14:23 PM] (19667) Starting service instance "<remote_folder>"
[Info  - 7:14:23 PM] (19667) Setting pythonPath for service "<workspace_root>": "/.../<workspace_root>/venv/bin/python"
[Info  - 7:14:23 PM] (19667) Loading configuration file at /.../<workspace_root>/pyrightconfig.json
[Info  - 7:14:23 PM] (19667) Assuming Python version 3.10
[Info  - 7:14:23 PM] (19667) Assuming Python platform Darwin
[Info  - 7:14:23 PM] (19667) No include entries specified; assuming /.../<workspace_root>
[Info  - 7:14:23 PM] (19667) Auto-excluding **/node_modules
[Info  - 7:14:23 PM] (19667) Auto-excluding **/__pycache__
[Info  - 7:14:23 PM] (19667) Auto-excluding **/.*
[Warn  - 7:14:23 PM] (19667) The useLibraryCodeForTypes has been specified in both the config file and the client settings. The value in the config file (true) will take precedence
[Warn  - 7:14:23 PM] (19667) stubPath /.../<workspace_root>/typings is not a valid directory.
[Info  - 7:14:24 PM] (19667) Search paths for /.../<workspace_root>
[Info  - 7:14:24 PM] (19667)   /Users/marc/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib
[Info  - 7:14:24 PM] (19667)   /.../<workspace_root>
[Info  - 7:14:24 PM] (19667)   /.../<workspace_root>/typings
[Info  - 7:14:24 PM] (19667)   /Users/marc/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stubs/...
[Info  - 7:14:24 PM] (19667)   /Users/marc/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/bundled/stubs
[Info  - 7:14:24 PM] (19667)   /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10
[Info  - 7:14:24 PM] (19667)   /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/lib-dynload
[Info  - 7:14:24 PM] (19667)   /.../<workspace_root>/venv/lib/python3.10/site-packages
[Info  - 7:14:24 PM] (19667)   /.../<workspace_root>/<project>
[Info  - 7:14:24 PM] (19667) Adding fs watcher for library directories:
 /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10
/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/lib-dynload
/.../<workspace_root>/venv/lib/python3.10/site-packages
[Info  - 7:14:24 PM] (19667) Adding fs watcher for directories:
 /.../<workspace_root>
[Info  - 7:14:24 PM] (19667) Searching for source files
[Info  - 7:14:31 PM] (19667) Auto-excluding /.../<workspace_root>/venv
[Info  - 7:14:32 PM] (19667) Found 19067 source files
...
```

> **Note**
> The issue seems to be related to these new / different lines
> ```diff
> +[Info  - 7:07:25 PM] (19522) Starting service instance "<default>"
> -[Info  - 7:14:24 PM] (19667) Search paths for /.../<workspace_root>
> +[Info  - 7:07:26 PM] (19522) Search paths for <default>
> ```

### Workspace layout
```
- <workspace_root>
  - venv
  - folder
    - python files
- <other_folder>
- <remote_folder>
```

## Environment data

-   Language Server version: 2023.2.12
-   OS and version: macOS 13
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10

**Comment by debonte** [comment]
> Error: spawn /.../<workspace_root>/venv/bin/python ENOENT

@rchiodo, my hunch is that this is caused by the `host.runScript` call in `PytestQueryEngine.getOptions`. Can you investigate?

**Comment by rchiodo** [comment]
Do you have a repro? I tried and couldn't repro the problem.

**Comment by rchiodo** [comment]
Maybe just having an invalid path as the venv?

**Comment by cdce8p** [comment]
> Do you have a repro? I tried and couldn't repro the problem.

It seems to be related to the remote folder. On macOS those will be mounted under `/volumes/...`. The issue occurs if the workspace contains a (remote) folder which isn't available. _The venv is still valid and reachable though._

I could reproduce it with these steps:
1. Create a new folder, e.g. `project`
2. Open `project` in new workspace
3. * Create Python virtual env: `python3.10 -m venv venv`
4.  Select Python interpreter -> `venv`
5.  Save Workspace As ...
6.  Edit workspace settings (JSON)
7.  Add currently unavailable folder path, e.g. `"/Volumes/test"`
8.  Reload workspace

**Comment by rchiodo** [comment]
Ah okay I can repro if I delete my venv after creating it.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.2.13, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023213-6-February-2023-PreRelease)

**Comment by cdce8p** [comment]
Thanks! Can confirm the issue is fixed in `2023.2.13`.
